lib/pbench/server/api/resources/intake_base.py

```diff
@@ -343,14 +343,14 @@ def _intake(
                         ofp.write(chunk)
                         hash_md5.update(chunk)
             except OSError as exc:
-                # NOTE: Werkzeug doesn't support status 509, so the abort call
-                # in _dispatch will fail. Rather than figure out how to fix
-                # that, just report as an internal error.
                 if exc.errno == errno.ENOSPC:
-                    msg = f"Out of space on {tar_full_path.root}"
-                else:
-                    msg = f"Unexpected error {exc.errno} encountered during file upload"
-                raise APIInternalError(msg) from exc
+                    raise APIAbort(
+                        HTTPStatus.REQUEST_ENTITY_TOO_LARGE,
+                        f"Out of space on {tar_full_path.parent}",
+                    )
+                raise APIInternalError(
+                    f"Unexpected error {exc.errno} encountered during file upload"
+                ) from exc
```

Comment:
I think part of the original motivation was to avoid "publishing" internal file paths. Since the introduction of `APIInternalError` that's not actually a problem, but I never bothered to change it.